scripts/msmarco-passage/create_faiss_index.py

```diff
@@ -0,0 +1,34 @@
+import argparse
+import os
+
+import faiss
+import numpy as np
+import pandas as pd
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--embedding", type=str, help='encoder name or path', required=True)
+    parser.add_argument("--index-type", type=str, help='doc file to get encoded.', required=True)
+    parser.add_argument("--index-dir", type=str, help='doc type, [passage, query]', required=True)
+    args = parser.parse_args()
+
+    print('Loading Embeddings')
+    embedding_df = pd.read_pickle(args.embedding)
```

Comment:
Oh, I think I see now... you pickle the encoded text, and then you run through here...

The corpus is big... 

I think your rationale is that you encode the corpus once, as pickled file, and then run through indexing script twice, once to generate flat index, the second time to generate hnsw index?

If that's the idea - how about the alternative: you encode and directly generate flat index. Then in a second pass you go over the flat index to generate the hnsw index. That way you don't need to materialize intermediate data in pickled form.

Reply:
good suggestion! we can always keep the brute force index as raw `embedding` data.
so we dont need the pickle for passage embedding. will do!

I worked on pickle because I feel it is easier to handle than tfrecord here.
For query embeddings, we create a single pickle file to store id and embeddings together? Maybe with the raw query text as well, since we may want to do the `text->embedding` map. I.e. `load_encoded_queries` load everyting from a pickle file

Reply:
see my comment below - yes, queries are small, pickle would be fine.

In general, I tend to avoid pickle for large amounts of data...

Actually, for reference - the corpus encoded as vectors - what's the size of picked data, FAISS flat index, and TF record?

Reply:
26G, 26G and 26G...

Reply:
Interesting. Pickled data isn't any larger.

I was always under the impression that Pickle wasn't very space efficient... guess I was wrong.